src/types.ts

```diff
@@ -42,6 +42,49 @@ export interface Ctx {
   _random?: {
     seed: string | number;
   };
+  // enhanced by events plugin
+  events?: EventsAPI;
+  // enhanced by player plugin
+  player?: PlayerAPI;
+  // enhanced by random plugin
+  random?: RandomAPI;
+}
+
+export interface EventsAPI {
```

Comment:
Let's move these to the relevant spots in `src/plugins`?

Reply:
You mean all in one `src/plugins/types.ts` file? Or somehow split for each plugin?

Reply:
@janKir It might be better if the interfaces are declared in the same place as the actual plugins (should make it easier to keep them in sync later), so perhaps each directly in the `plugin-events` and `plugin-random` files? Then they can be imported in `src/types`.